connectorx-python/connectorx/tests/test_mysql.py

```diff
@@ -20,80 +20,190 @@ def test_mysql_without_partition(mysql_url: str) -> None:
+    )
+    expected = pd.DataFrame(
+        index=range(6),
+        data={
+            "test_int": pd.Series([1, 2, 3, 4, 5, 6], dtype="Int64"),
+            "test_float": pd.Series([1.1, 2.2, 3.3, 4.4, 5.5, 6.6], dtype="float64"),
+            "test_enum": pd.Series(['odd', 'even', 'odd', 'even', 'odd', 'even'], dtype="object"),
+            "test_null": pd.Series([None, None, None, None, None, None], dtype="Int64")
+        }
+    )
+    assert_frame_equal(df, expected, check_names=True)
+
+
+def test_read_sql_with_partition_without_partition_range(mysql_url: str) -> None:
+    query = "SELECT * FROM test_table where test_float > 3"
     df = read_sql(
         mysql_url,
         query,
         partition_on="test_int",
         partition_num=3,
     )
+
+    expected = pd.DataFrame(
+        index=range(4),
+        data={
+            "test_int": pd.Series([3, 4, 5, 6], dtype="Int64"),
+            "test_float": pd.Series([3.3, 4.4, 5.5, 6.6], dtype="float64"),
+            "test_enum": pd.Series(['odd', 'even', 'odd', 'even'], dtype="object"),
+            "test_null": pd.Series([None, None, None, None], dtype="Int64")
+        }
+    )
+    assert_frame_equal(df, expected, check_names=True)
+
+
+def test_mysql_manual_partition(mysql_url: str) -> None:
+    queries = [
+        "SELECT * FROM test_table WHERE test_int < 2",
+        "SELECT * FROM test_table WHERE test_int >= 2",
+    ]
+    df = read_sql(
+        mysql_url,
+        query=queries)
     expected = pd.DataFrame(
         index=range(6),
         data={
             "test_int": pd.Series([1, 2, 3, 4, 5, 6], dtype="Int64"),
-            "test_float": pd.Series([1.1, 2.2, 3.3, 4.4, 5.5, 6.6], dtype="float64")
+            "test_float": pd.Series([1.1, 2.2, 3.3, 4.4, 5.5, 6.6], dtype="float64"),
+            "test_enum": pd.Series(['odd', 'even', 'odd', 'even', 'odd', 'even'], dtype="object"),
+            "test_null": pd.Series([None, None, None, None, None, None], dtype="Int64"),
         }
     )
     assert_frame_equal(df, expected, check_names=True)
 
 
-def test_mysql_types(mysql_url: str) -> None:
-    query = "select * from test_types"
-    df = read_sql(mysql_url, query)
+def test_mysql_selection_and_projection(mysql_url: str) -> None:
+    query = "SELECT test_int FROM test_table WHERE test_float < 5"
+    df = read_sql(
+        mysql_url,
+        query,
+        partition_on="test_int",
+        partition_num=3,
+    )
+    expected = pd.DataFrame(
+        index=range(4),
+        data={
+            "test_int": pd.Series([1, 2, 3, 4], dtype="Int64"),
+        }
+    )
+    assert_frame_equal(df, expected, check_names=True)
+
+
+def test_mysql_join(mysql_url: str) -> None:
+    query = "SELECT T.test_int, T.test_float, S.test_str FROM test_table T INNER JOIN test_table_extra S ON T.test_int = S.test_int"
+    df = read_sql(
+        mysql_url,
+        query,
+        partition_on="test_int",
+        partition_num=3,
+    )
+    print(df)
```

Comment:
It looks like there are several debug `print(df)` accidentally left in tests. Can you help to remove all of them?